ide/docker/README.md

```diff
@@ -47,3 +48,45 @@ docker run -it --rm -v $PWD:/opt/xcsoar:delegated xcsoar-build xcsoar-compile AN
 Note: the `delegated` volume modifier is used to speed up compilation on non Linux hosts 
 (macOS and Windows), it allows for eventual consistent filesystem change propagation 
 from the container to the host system.
+
+
+## Instructions for running XCSoar within the Docker container
+
+You can also use the Docker container to run the Linux build of XCSoar on a non-Linux
+system. This could be helpful to perform quick tests. It will use software rendering.
+Needless to say that the experience is inferior to running XCSoar on the host
+system directly. However, installing all build dependencies might also be overkill
```

Comment:
You don't need to install build dependencies to run XCSoar. Only the shared libraries XCSoar was linked with. Which most users probably already have.

Reply:
If you are running macOS or Windows you need a whole build environment, such as compilers.

Reply:
Not for running.

Reply:
Ah, now I got your point. Sure thing, but the target audience here is someone wanting to make (small) changes to XCSoar and therefore needing a way to compile it also.

I was in this situation some time ago and after spending some time failing to get it to work natively on macOS. I decided it would be easier to just let it run on the Docker container.

Reply:
Sure they can compile it in the container, but the point of this PR is to *run* it in the container. You can compile in the container which has all the development tools installed, but is running inside the container really worth it?
It is dangerous and complex and error prone and slow; for example the user has to disable authentication in his X server.
(I'm not convinced, but that doesn't mean anything - I'm not convinced of Docker at all, that's why I have never used it.)

Reply:
Is it possible to mount in the Xauthority file on OSX just like in the medium.com example? that would make it safer for the enduser.

@MaxKellermann docker is dead. there are much saner run-times for running containers now called cri-o and podman as the pulling tool. Its a challenge to run/compile master on pretty much anything that isn't debian buster or a direct target.

Reply:
@sandrinr in theory one cloud build a runtime container only, or one only containing the unix target. It would be much smaller than this multigigabyte monster with all the toolchains. But for that kind of runtime I'd prefeer snapper or static binaries.

Reply:
@MaxKellermann You got a point. However, AFAIK it is not possible to build macOS binaries on Linux. I think you need Xcode for that which is only available on macOS. So building the Linux binaries and running them within the Linux container is the whole point. Besides, in XCSoar history the Linux target always worked well. Not sth that could be told about the OSX target.
However, I do agree the approach being complicated (re-routing of X11 traffic, ...). It was just simpler for me as opposed to fixing the OSX target. I guess there is no harm in deciding that this should not be part of XCSoar.

Reply:
@lordfolken Docker isn't dead, that's simply not true. It's used *a lot* all over the world. Even if there are better ones, they haven't hit mainstream use yet, and with that in mind having a docker container is better. It wouldn't hurt having others too, but let's not skip the docker one just because it's docker :)

Reply:
So this is about building a Linux binary on Mac OS X and then running the Linux binary with Docker on Mac OS X? Does that even work? I thought Docker is a container manager, and containers share the host kernel, which isn't Linux... is Docker a VM on Mac OS X?

Reply:
@MaxKellermann correct docker on os/x and windows use virtualization with very lightweight OSes (ramdisk/kernel)

Reply:
Docker on macOS uses xhyve (practically bhyve) to run a VM one can enter into using `screen ~/Library/Containers/com.docker.docker/Data/vms/0/tty` to poke around:

~~~
docker-desktop:~# uname -a
Linux docker-desktop 4.19.76-linuxkit #1 SMP Thu Oct 17 19:31:58 UTC 2019 x86_64 Linux
docker-desktop:~# cat /etc/os-release 
PRETTY_NAME="Docker Desktop"
~~~